### PR TITLE
Move duplicated files in asn-supl/ and asn-rrl/ to a new types/ subdir

### PR DESCRIPTION
**Pull request by orlando017**
Signed-off-by: Will Silberman [wsilberm@google.com](mailto:wsilberm@google.com)

Move duplicated files in asn-supl/ and asn-rrl/ to a new types/ subdir. Note that in some cases, there are slight differences between the versions from the two folders, such as function argument names or in some cases data types; eg here's the diff between asn-supl/INTEGER.c and asn-rrlp/INTEGER.c:

```
54c54
<      * (Remove too int64_t sign extension, remove some first 0x00 bytes)
---
>      * (Remove too long sign extension, remove some first 0x00 bytes)
226c226
<     /* Output in the int64_t xx:yy:zz... format */
---
>     /* Output in the long xx:yy:zz... format */
421c421
<     int64_t sign = 1LL;
---
>     int64_t sign = 1;
1057c1057
<              * able to >sizeof(int64_t) bytes,
---
>              * able to >sizeof(long) bytes,
1085c1085
<                     /* Still cannot fit the int64_t */
---
>                     /* Still cannot fit the long */
```
I suspect such changes are inadvertent from after the code was duplicated. Per jwmelto's suggestion in https://github.com/gnss-sdr/gnss-sdr/pull/599, I'm prioritizing the file versions in asn-supl, which. Ibelieve are all explicitly sized types.

Having the duplicated symbols is error prone, and causes issues when linking with some compilers.

I tested this by building and running gnss-sdr per https://gnss-sdr.org/my-first-fix/, and the program ran fine as far as I could tell.

### CODE REVIEW

**Comment by carlesfernandez** [comment]
Dear @orlando017,

It seems that `src/core/libs/supl/types/*` is missing.

I like the idea of this PR, though. That code was generated automatically long time ago and then retouched to make modern compilers and linters happier, but it still does its job and retrieves A-GPS data from the net.  Since it will not be regenerated again (at least for that SUPL version), I'm in favor of making those changes if they allow more portability. 

For the duplicated files with small differences, I would retain those with `int64_t ` instead of `long`. We overlooked that.

**Comment by orlando017** [comment]
Thank you! Fixed, added back the files in types/. All tests except for clang-tidy are now succeeding; I'm not sure what's going wrong there, from the 'checks' tab it looks like the check is still running.

Is there a way to restart the clang tidy check (or all checks)?

File: src/core/libs/supl/CMakeLists.txt
Comment (jwmelto) [inline src/core/libs/supl/CMakeLists.txt:65]: This change is consistent with the surrounding lines, however there is an opportunity to improve all of them (I usually fix too much, so feel free to ignore this). A more maintainable change would be 
```suggestion
        ${CMAKE_CURRENT_SOURCE_DIR}/types
```

File: src/core/libs/supl/CMakeLists.txt
Comment (jwmelto) [inline src/core/libs/supl/CMakeLists.txt:70]: Similarly, consider
```suggestion
    INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR};${CMAKE_CURRENT_SOURCE_DIR}/asn-supl;${CMAKE_CURRENT_SOURCE_DIR}/asn-rrlp;${CMAKE_CURRENT_SOURCE_DIR}/types;${GNUTLS_INCLUDE_DIR}"
```

Now, I'd have to prove this out, but I'm pretty sure that this is equivalent (and more maintainable):
```suggestion
    INTERFACE_INCLUDE_DIRECTORIES 
        ${CMAKE_CURRENT_SOURCE_DIR}
        ${CMAKE_CURRENT_SOURCE_DIR}/asn-supl
        ${CMAKE_CURRENT_SOURCE_DIR}/asn-rrlp
        ${CMAKE_CURRENT_SOURCE_DIR}/types
        ${GNUTLS_INCLUDE_DIR}
```

File: src/core/libs/supl/CMakeLists.txt
Comment (jwmelto) [inline src/core/libs/supl/CMakeLists.txt:15]: As a general rule, CMake recommends _against_ using **glob** patterns. The one place where they are appropriate is with generated files. As @carlesfernandez kindly pointed out, this code was originally generated, which explains the pattern here.

If we are now moving to explicit maintenance, would it be better to explicitly enumerate the sources and their headers? I don't feel strongly about this, but hope that the maintainers will have an opinion.

**Review by jwmelto — APPROVED** [review]
Thanks for using proper **git** practices. It made this so much easier to review. Given that you are literally moving one set of files to another directory and deleting the duplicate files, I have no issues with this PR. For what that's worth.

File: src/core/libs/supl/CMakeLists.txt
Comment (carlesfernandez) [reply src/core/libs/supl/CMakeLists.txt:65]: That's nice, I will apply that before merging.

File: src/core/libs/supl/CMakeLists.txt
Comment (orlando017) [reply src/core/libs/supl/CMakeLists.txt]: Done.

File: src/core/libs/supl/CMakeLists.txt
Comment (orlando017) [reply src/core/libs/supl/CMakeLists.txt]: I'd rather leave as is for now if that's ok. It would be a very long list to write out.

**Comment by carlesfernandez** [comment]
Thanks! Merging this PR into the `next` branch.

**Pull request closed without merging**
